challenges.yml

```
@@ -193,3 +193,245 @@ stimulus_values:
+  explanation: |
+    The "Value Change Callback" feature of the Stimulus values API implements a naming convention. The method's name
+    must be equal to the name of the *value* property plus the word `ValueChanged`.
+  features:
+    multiple_choice:
+      choices:
+        a: |
+          ```js
+          itemIdChanged() {
+              ...
+          }
+          ```
+        b: |
+          ```js
+          itemIdValueChanged() {
+              ...
+          }
+          ```
+        c: |
+          ```js
+          onChangeItemIdValue() {
+              ...
+          }
+          ```
+        d: |
+          ```js
+          itemIdValue() {
+              ...
+          }
+          ```
+      correct: b
+search_preview:
+  question: |
+    How can you pass data to a Stimulus controller from a Twig template using the Values API?
+  explanation: |
+    The second argument of Encore's built-in function, the `stimulus_controller` function, helps us on that,
+    you only have to pass an object where the key name matches to one of the defined `values` in a controller.
+  features:
+    multiple_choice:
+      choices:
+        a: |
+          ```twig
+          {{ stimulus_controller('search-preview', {
+              id: object.id
+          }) }}
+          ```
+        b: |
+          ```twig
+          {{ controller_stimulus('search-preview', {
+              id: object.id
+          }) }}
+          ```
+        c: |
+          ```twig
+          {{ stimulus_controller_value('search-preview', {
+              id: object.id
+          }) }}
+          ```
+      correct: a
+stimulus_use:
+  question: |
+    Suppose you have this Stimulus controller which enables the "click outside" functionality. What is it missing?
+
+    ```js
+    import { Controller } from 'stimulus';
+    import { useClickOutside  } from 'stimulus-use';
+
+    export default class extends Controller {
+        static values = {
+            url: String,
+        }
+
+        clickOutside(event) {
+            // some interesting code here...
+        }
+    }
+    ```
+  explanation: |
+    In order to tell Stimulus you want to enable the `useClickOutside` feature, besides importing the module you need to
+    initialize it in the `connect()` method, e.g.
+
+    ```js
+    connect() {
+        useClickOutside(this);
+    }
+    ```
+  features:
+    multiple_choice:
+      choices:
+        a: |
+          The `clickOutside()` method should be prefixed with `async`
+        b: |
+          The `useClickOutside` feature must be initialized in the `connect()` method
+        c: |
+          The controller is missing a target.
+        d: |
+          Nothing.
+      correct: b
+multi_controller_dispatch:
+  question: |
+    How can I pass the `id` of an item from one to another Stimulus controller?
+  explanation: |
+    The `stimulus-use` library comes with the `useDispatch` behavior which helps with dispatching and listening to events.
```

```suggestion
    The `stimulus-use` library comes with a `useDispatch` behavior which helps with dispatching.
```

(it doesn't help with the listening side)